webpack/components/Table/TableHooks.js

```diff
@@ -41,14 +42,151 @@ class ReactConnectedSet extends Set {
       this.delete(id);
     }
   }
+
+  addAll(ids) {
+    ids.forEach(id => super.add(id));
+    this.forceRender();
+  }
 }
 
-const useSet = (initialArry) => {
+export const useSet = (initialArry) => {
   const [, setToggle] = useState(0);
   // needed because mutating a Ref won't cause React to rerender
   const forceRender = () => setToggle(prev => prev + 1);
   const set = useRef(new ReactConnectedSet(initialArry, forceRender));
   return set.current;
 };
 
-export default useSet;
+export const useSelectionSet = (results, metadata, initialArry = [], idColumn = 'id') => {
+  const selectionSet = useSet(initialArry);
+  const ids = results?.map(result => result[idColumn]) ?? [];
+  const areAllRowsOnPageSelected = () => Number(ids?.length) > 0 &&
+                                         ids.every(result => selectionSet.has(result));
+
+  const areAllRowsSelected = () =>
+    Number(selectionSet.size) > 0 && selectionSet.size === Number(metadata.total);
+
+  const selectPage = () => selectionSet.addAll(ids);
+  const selectNone = () => selectionSet.clear();
+  const onRowSelect = (isSelected, id) => {
+    if (isSelected) {
+      selectionSet.add(id);
+    } else {
+      selectionSet.delete(id);
+    }
+  };
+
+  const selectedCount = selectionSet.size;
+
+  const isSelected = id => selectionSet.has(id);
+
+  return {
+    onRowSelect,
+    selectedCount,
+    areAllRowsOnPageSelected,
+    areAllRowsSelected,
+    selectPage,
+    selectNone,
+    isSelected,
+    selectionSet,
+  };
+};
+
+export const useBulkSelect = (results, metadata, initialArry = [], idColumn = 'id') => {
+  const { selectionSet: inclusionSet, ...selectOptions } =
+                useSelectionSet(results, metadata, initialArry, idColumn);
+  const exclusionSet = useSet([]);
+  const [searchQuery, updateSearchQuery] = useState('');
+  const [isSelectAllMode, setSelectAll] = useState(false);
```

Comment:
The naming is closer but let's standardize it a bit :)
```suggestion
  const [selectAllMode, setSelectAllMode] = useState(false);
```